### ISSUE TITLE
Unbound variable: scheme-report-environment

### ISSUE THREAD

**Issue by BookOwl** [Jonathan50/do-it#15]
```
$ guile compile.scm < fib.di > fib.s
;;; note: auto-compilation is enabled, set GUILE_AUTO_COMPILE=0
;;;       or pass the --no-auto-compile argument to disable.
;;; compiling /Users/matthew/Desktop/do-it/compile.scm
;;; /Users/matthew/Desktop/do-it/compile.scm:435:39: warning: possibly unbound variable `scheme-report-environment'
;;; compiled /Users/matthew/.cache/guile/ccache/2.0-LE-8-2.0/Users/matthew/Desktop/do-it/compile.scm.go
Backtrace:
In ice-9/boot-9.scm:
 157: 10 [catch #t #<catch-closure 108a39580> ...]
In unknown file:
   ?: 9 [apply-smob/1 #<catch-closure 108a39580>]
In ice-9/boot-9.scm:
  63: 8 [call-with-prompt prompt0 ...]
In ice-9/eval.scm:
 432: 7 [eval # #]
In ice-9/boot-9.scm:
2401: 6 [save-module-excursion #<procedure 1087b3a00 at ice-9/boot-9.scm:4045:3 ()>]
4052: 5 [#<procedure 1087b3a00 at ice-9/boot-9.scm:4045:3 ()>]
In unknown file:
   ?: 4 [load-compiled/vm "/Users/matthew/.cache/guile/ccache/2.0-LE-8-2.0/Users/matthew/Desktop/do-it/compile.scm.go"]
In /Users/matthew/Desktop/do-it/compile.scm:
 460: 3 [compile-program (begin # #) #<output: file 1>]
In ice-9/boot-9.scm:
 867: 2 [call-with-input-file "library.do-it" ...]
 768: 1 [for-each #<procedure 1090a25a0 at /Users/matthew/Desktop/do-it/compile.scm:261:3 (x)> ...]
In /Users/matthew/Desktop/do-it/compile.scm:
 435: 0 [compile-defmacro (defmacro inc (var) ...) #<output: file 1> ...]

/Users/matthew/Desktop/do-it/compile.scm:435:39: In procedure compile-defmacro:
/Users/matthew/Desktop/do-it/compile.scm:435:39: In procedure module-lookup: Unbound variable: scheme-report-environment
```

**Comment by Jonathan50** [comment]
Oh yeah.  Guile doesn't have R5RS' `scheme-report-environment`, so (something like) `(use-modules (ice-9 r5rs))` needs to be evaluated for macros to work.

In MIT/GNU Scheme you can do

```
(define (scheme-report-environment version) user-initial-environment)
```

I should probably mention this in the README.

**Comment by Jonathan50** [comment]
It was fixed with c83a171a670ea3f72dc6c009b46778a256f59ddf
